components/signoffs/signoffDao.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const { db, pgp, NotFoundError } = require('../../utils/dbUtils');
+const logger = require('../../utils/logger').logger;
+const queryStream = require('pg-query-stream');
+const qrec = pgp.errors.queryResultErrorCode;
+
+const id = 'id';
+const created = 'created';
+const caseIdentifier = 'case_identifier';
+const username = 'username';
+const qcPassed = 'qc_passed';
+const signoffStepName = 'signoff_step_name';
+const deliverableType = 'deliverable_type';
+
+const signoffCols = [
+  id,
+  created,
+  caseIdentifier,
+  username,
+  qcPassed,
+  signoffStepName,
+  deliverableType,
+];
+
+const signoffColsCreate = new pgp.helpers.ColumnSet(
+  [caseIdentifier, username, qcPassed, signoffStepName, deliverableType],
+  { table: 'signoff' }
+);
+
+const addSignoff = (signed) => {
+  return new Promise((resolve, reject) => {
+    db.task('add-signoffs', async (tx) => {
+      const signoffData = {
+        case_identifier: signed.caseIdentifier,
+        username: signed.username,
+        qc_passed: signed.qcPassed,
+        signoff_step_name: signed.signoffStepName,
+        deliverable_type: signed.deliverableType,
+      };
+
+      const signoffInsert = pgp.helpers.insert(
+        signoffData,
+        signoffColsCreate,
+        'signoff'
+      );
+      //add const onConflict here ?
```

Comment:
Yes, do add `onConflict` here, since you want behaviour to be "If a sign-off is added with the same caseIdentifier, signoffStepName, username, and deliverableType the old sign-off record will be replaced (regardless of QC status)" rather than "error that this already exists"